### PR TITLE
Generate return from Load File into REPL so Alive will refresh Tree View properly

### PR DESCRIPTION
**Pull request by madkins23**
I'm sending this not so much as a PR as documentation of something I found yesterday.

TLDR; With this fix the Packages and ASDF Systems views in the Lisp Tree View will update automatically after a file load.

I noticed in the code for Alive that after loading a file into the REPL there was code to refresh packages and ASDF systems in the Lisp Tree View. I did some debugging and found that the load file call never returned so the subsequent calls never executed.

I used configuration #2 in the Alive-lsp dev doc I submitted earlier and after muddling through I was able to find what I thought of as a "fix" though I don't understand all of what's going on there. I verified it this morning using configuration #3 (which fixed a few issues in the dev doc).

It's a simple fix but I wonder if the issue might not be upstream in `run-in-thread` which I do not yet understand. There is a statement:
```
(send-msg state (notification:refresh))
```
in that function which may be intended to tell the client to do a refresh but I haven't figured out how that would work.

In the meantime I found this fix by looking at `handle-load-asdf` which is where I got the other fix (which just returns `t`) that is commented out in this PR. So this may just be a case of not writing the two functions according to the same template.

### CODE REVIEW

**Comment by madkins23** [comment]
Without fix:
```
17:02:17 INF LSP starting
17:02:17 INF LSP client starting.
17:02:17 DBG Send msg={"id":81,"jsonrpc":"2.0","method":"$/alive/loadFile","params":{"path":"/home/marc/work/go/src/github.com/madkins23/lsp-tester/msgs/alive/simple.lisp","showStderr":true,"showStdout":true}}
17:02:17 DBG Received msg={"jsonrpc":"2.0","method":"$/alive/refresh","params":{}}
17:02:17 DBG Received msg={"jsonrpc":"2.0","method":"$/alive/stdout","params":{"data":"; compiling file \"/home/marc/work/go/src/github.com/madkins23/lsp-tester/msgs/alive/simple.lisp\" (written 05 MAY 2023 04:54:19 PM):"}}
17:02:17 DBG Received msg={"jsonrpc":"2.0","method":"$/alive/stdout","params":{"data":"; wrote /home/marc/work/go/src/github.com/madkins23/lsp-tester/msgs/alive/simple.fasl"}}
17:02:17 DBG Received msg={"jsonrpc":"2.0","method":"$/alive/stdout","params":{"data":"; compilation finished in 0:00:00.004"}}
17:02:17 DBG Received msg={"jsonrpc":"2.0","method":"$/alive/stdout","params":{"data":"5040 "}}
17:02:17 DBG Received msg={"jsonrpc":"2.0","method":"$/alive/stdout","params":{"data":"\"Hello World!\" "}}
17:02:17 DBG Received msg={"jsonrpc":"2.0","method":"$/alive/refresh","params":{}}
```

With fix:
```
17:02:58 INF LSP starting
17:02:58 INF LSP client starting.
17:02:58 DBG Send msg={"id":81,"jsonrpc":"2.0","method":"$/alive/loadFile","params":{"path":"/home/marc/work/go/src/github.com/madkins23/lsp-tester/msgs/alive/simple.lisp","showStderr":true,"showStdout":true}}
17:02:58 DBG Received msg={"jsonrpc":"2.0","method":"$/alive/refresh","params":{}}
17:02:58 DBG Received msg={"jsonrpc":"2.0","method":"$/alive/stdout","params":{"data":"; compiling file \"/home/marc/work/go/src/github.com/madkins23/lsp-tester/msgs/alive/simple.lisp\" (written 05 MAY 2023 04:54:19 PM):"}}
17:02:58 DBG Received msg={"jsonrpc":"2.0","method":"$/alive/stdout","params":{"data":"; wrote /home/marc/work/go/src/github.com/madkins23/lsp-tester/msgs/alive/simple.fasl"}}
17:02:58 DBG Received msg={"jsonrpc":"2.0","method":"$/alive/stdout","params":{"data":"; compilation finished in 0:00:00.004"}}
17:02:58 DBG Received msg={"jsonrpc":"2.0","method":"$/alive/stdout","params":{"data":"5040 "}}
17:02:58 DBG Received msg={"jsonrpc":"2.0","method":"$/alive/stdout","params":{"data":"\"Hello World!\" "}}
17:02:58 DBG Received msg={"id":81,"jsonrpc":"2.0","result":{"messages":null}}
17:02:58 DBG Received msg={"jsonrpc":"2.0","method":"$/alive/refresh","params":{}}
```
The second to the last line is the result message that isn't being emitted by the LSP right now. That would likely hold up the client code from ever returning.

File: src/session.lisp [hunk 1]
```diff
@@ -570,7 +570,8 @@
                                                             :stderr-fn (lambda (data)
                                                                            (when (assoc :show-stderr params)
                                                                                  (send-msg state (notification:stderr data)))))))
-
+                                     ;  (send-msg state (message:create-response id :result-value T))
+                                     (send-msg state (resp:load-file id msgs))
```
Comment (nobody-famous) [inline src/session.lisp:574]: It needs to just be this line. resp:load-file creates the load file response message. Apparently when I moved this to run in a thread, I forgot to wrap that in send-msg so it would send the response. The handlers that don't run in a thread return their responses, which get sent in read-messages.

**Comment by nobody-famous** [comment]
The refresh notification in run-in-thread tells the client to refresh it's display. It might not be in exactly the right spot. unwind-protect is the equivalent of try..finally in other languages. I just pushed a fix for the parens being out of whack. The idea is that whenever a thread is done, the client needs to refresh in case the function updated something important. User initiated operations, like eval and load, run in their own thread so they don't end up blocking the whole server. Everything else runs sequentially because the VSCode messages, like file updates, need to be done in order.

**Comment by madkins23** [comment]
Cleaned up the submission so you can take it as is if you want. If you're rolling the fix into some other work that you're already doing then you can just close this PR.
